data/crac-creation/crac-creator-csa-profiles/src/main/java/com/powsybl/openrao/data/craccreation/creator/csaprofile/craccreator/cnec/FlowCnecInstantHelper.java

```diff
@@ -0,0 +1,115 @@
+        checkCsaExtension(csaParameters);
+        checkUsePatlInFinalStateMap(csaParameters);
+        checkCraApplicationWindowMap(csaParameters);
+        tsosWhichDoNotUsePatlInFinalState = csaParameters.getUsePatlInFinalState().entrySet().stream().filter(entry -> !entry.getValue()).map(Map.Entry::getKey).collect(Collectors.toSet());
+        curative1InstantDuration = csaParameters.getCraApplicationWindow().get(CURATIVE_1_INSTANT);
+        curative2InstantDuration = csaParameters.getCraApplicationWindow().get(CURATIVE_2_INSTANT);
+        curative3InstantDuration = csaParameters.getCraApplicationWindow().get(CURATIVE_3_INSTANT);
+    }
+
+    public Set<String> getTsosWhichDoNotUsePatlInFinalState() {
+        return tsosWhichDoNotUsePatlInFinalState;
+    }
+
+    // CSA CRAC Creation Parameters checking
+
+    private static void checkCsaExtension(CsaCracCreationParameters csaParameters) {
+        if (csaParameters == null) {
+            throw new OpenRaoException("No CsaCracCreatorParameters extension provided.");
+        }
+    }
+
+    private void checkUsePatlInFinalStateMap(CsaCracCreationParameters csaParameters) {
+        Map<String, Boolean> usePatlInFinalState = csaParameters.getUsePatlInFinalState();
+        for (String tso : tsos) {
+            if (!usePatlInFinalState.containsKey(tso)) {
+                throw new OpenRaoException("use-patl-in-final-state map is missing \"" + tso + "\" key.");
+            }
+        }
+    }
+
+    private void checkCraApplicationWindowMap(CsaCracCreationParameters csaParameters) {
+        Map<String, Integer> craApplicationWindow = csaParameters.getCraApplicationWindow();
+        for (String instant : instants) {
+            if (!craApplicationWindow.containsKey(instant)) {
+                throw new OpenRaoException("cra-application-window map is missing \"" + instant + "\" key.");
+            }
+        }
+        if (craApplicationWindow.get(CURATIVE_1_INSTANT) > craApplicationWindow.get(CURATIVE_2_INSTANT)) {
+            throw new OpenRaoException("The TATL acceptable duration for %s cannot be longer than the acceptable duration for %s.".formatted(CURATIVE_1_INSTANT, CURATIVE_2_INSTANT));
+        }
+        if (craApplicationWindow.get(CURATIVE_2_INSTANT) > craApplicationWindow.get(CURATIVE_3_INSTANT)) {
+            throw new OpenRaoException("The TATL acceptable duration for %s cannot be longer than the acceptable duration for %s.".formatted(CURATIVE_2_INSTANT, CURATIVE_3_INSTANT));
+        }
+    }
+
+    // Instant to limits mapping
+
+    Set<Integer> getAllTatlDurationsOnSide(Branch<?> branch, TwoSides side) {
+        return branch.getCurrentLimits(side).map(limits -> limits.getTemporaryLimits().stream().map(LoadingLimits.TemporaryLimit::getAcceptableDuration).collect(Collectors.toSet())).orElseGet(Set::of);
+    }
+
+    public Map<String, Integer> mapPostContingencyInstantsAndLimitDurations(Branch<?> branch, TwoSides side, String tso) {
```

Comment:
I don't understand why limits are min tatl greater than each instantDuration, and not max tatl shorter than each instantDuration ? I need an explanation

Reply:
The main thing to remember (and the source of many problems and confusions) is the fact that a TATLX means that the flow can be up to this value for X seconds, and after those X seconds, it has to be lower than the next TATL.

So for instance if you have a TATL300 and a TATL900, you can be up to the TATL300 for the first 300 seconds, then up to TATL900 for the next 600s (so up to 900s after the contingency happened), and then back to the PATL. (keep in mind TATL300 >= TATL900 >= PATL)

If the first curative batch happens 300s after the contingency, and the second batch 600s after the contingency, and the third batch 1200s after the contingency:
- if we check the TATL300 in the first batch, then we would allow the flow to be at the TATL300 for all the time from 300s to 600s. This would mean that we would be allowed to exceed the TATL900 from t+300 to t+600 but we should have been under it.
- if we check the TATL900 in the first batch, we will correctly limit the flow to TATL900 from t+300 to t+600. And then for the next batch we will have to check the PATL already for the same reasons (we need to be under the TATL900 from t+600 to t+900 and under the PATL after t+900, but the third batch is applied at t+1200 so we already need to be under PATL after the second batch)